clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -976,12 +976,46 @@ bool tools::addOpenMPRuntime(ArgStringList &CmdArgs, const ToolChain &TC,
   return true;
 }
 
-void tools::addFortranRuntimeLibs(const ToolChain &TC,
+void tools::addFortranRuntimeLibs(const ToolChain &TC, const ArgList &Args,
                                   llvm::opt::ArgStringList &CmdArgs) {
   if (TC.getTriple().isKnownWindowsMSVCEnvironment()) {
-    CmdArgs.push_back("Fortran_main.lib");
-    CmdArgs.push_back("FortranRuntime.lib");
-    CmdArgs.push_back("FortranDecimal.lib");
+    CmdArgs.push_back(Args.MakeArgString(
+        "/DEFAULTLIB:" + TC.getCompilerRTBasename(Args, "builtins")));
+    unsigned RTOptionID = options::OPT__SLASH_MT;
```

Comment:
As of LLVM 17.0's distribution, the Fortran runtime libraries are built with `msvcrt`, so I think the current default is actually `/MD`.  Since this wasn't really modeled before, and CMake will be taught to pass one of the four flags explicitly anyway, changing the default may not matter, but it's something to be aware of.

Reply:
In local testing, `flang-new -fms-runtime-lib=static foo.f90 -v`, where `foo.f90` is an empty `program` statement, fails with a bunch of unresolved CRT symbols.

Reply:
Could you share the missing symbols? If it's something like `__udivti3` then that's an issue with not finding compiler-rt (those symbols are for 128 bit types and not contained in the MSVC rt libs of any flavour), see https://github.com/llvm/llvm-project/issues/25679. If so, there's perhaps a chance that things work with `flang-new -DAVOID_NATIVE_UINT128_T=1`...

Reply:
I should be adding the missing compiler-rt libs in this patch too. I don’t see any missing symbols in any of the configurations when testing an empty program locally. Could you share the error you’re seeing?

Reply:
The errors were due to https://github.com/llvm/llvm-project/pull/70833#pullrequestreview-1710341215 because the runtime library variants not being built with the correct CRT themselves.  After switching back to the `CMAKE_MSVC_RUNTIME_LIBRARY` the problem is resolved.